### PR TITLE
[FEATURE] Added math constants to simc

### LINKED ISSUE #202: [FEATURE] Add constants to sim-c
**The Problem**
I have realized that sim-c contains no constants(PI, TAU, etc.)
This makes it rather inconvenient to do any sort of math that involves circles and you have to manually type in the digits of your constant, which is both tedious and error-prone.

**The solution**
Implement in sim-c a way for you to be able to call constants directly, where on the backend this would convert to the math.h equivalent. For example, writing PI in sim-c would be converted to M_PI in C code(And import math.h).

**Current Workaround**
When I wrote a program to calculate the area of a sphere, I was able to take advantage of the fact that I can declare and initialize a variable in sim-c, then I would initialize that variable to its math.h equivalent constant in C code, and then I was able to use the constant. See /simc-codes/q73.simc. This is a very long and tedious task for something that is already pre built in C's math.h.

I'd be happy to work on this if you approve of it.

I look forward to your response.

### PR DESCRIPTION
**Pull request by lucasace**
Added the following math constants to simc
Fixes #202 
```
C Name->SIMC Name
M_PI ->PI
M_E -> E
INFINITY -> inf
NAN ->NaN
```

Test CODE:
```simc
MAIN
	print(inf)
	print(E)
	print(PI)
	print(NaN)
	print(-inf)
END_MAIN
```

### CODE REVIEW

**Review by frankhart2018 — CHANGES_REQUESTED** [review]
Looks good, I just wanted to point out that keeping the constants all in upper case (like in C) will be better otherwise someone might feel they are just variables.

**Comment by lucasace** [comment]
How about PI,E,inf,NaN

**Comment by frankhart2018** [comment]
Yeah this looks good too @lucasace. Have you made the changes?

**Comment by lucasace** [comment]
> Yeah this looks good too @lucasace. Have you made the changes?

Yes, I have made the changes

File: simc/lexical_analyzer.py [hunk 1]
```diff
@@ -189,6 +189,12 @@ def keyword_identifier(source_code, i, table, line_num):
         value += source_code[i]
         i += 1
 
+    #Check if value is a math constant or not
+    if value in ["PI","E","inf","NaN"]:
```
Comment (lucasace) [inline simc/lexical_analyzer.py:193]: Yes I have made the changes

**Review by frankhart2018 — APPROVED** [review]
Looks good, merging this. Thanks for your contribution @lucasace. Closes #202.

**Pull request merged**